eth2/operation_pool/src/attestation.rs

```diff
@@ -40,15 +58,18 @@ impl<'a, T: EthSpec> MaxCover for AttMaxCover<'a, T> {
     fn update_covering_set(
         &mut self,
         best_att: &Attestation<T>,
-        covered_validators: &BitList<T::MaxValidatorsPerCommittee>,
+        covered_validators: &HashMap<u64, u64>,
     ) {
         if self.att.data.slot == best_att.data.slot && self.att.data.index == best_att.data.index {
-            self.fresh_validators.difference_inplace(covered_validators);
+            for key in covered_validators.keys() {
+                let _ = self.fresh_validators_rewards.remove(key);
+            }
```

Comment:
1. Is this correct?
2. Is there a more efficient way of doing this? Using `let _` because we don't want clippy to warn us of unused result and we don't want to `unwrap` because sometimes I believe we'll get `None`...

Reply:
1. Yep!
2. Yeah the `let _` is a bit of an anti-pattern. We could use [`HashMap::retain`](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.retain) here instead of the loop.